tests/test_marquez_client.py

```diff
@@ -258,6 +258,77 @@ def test_create_job(self, mock_put):
         assert str(response['id']) is not None
         assert str(response['location']) == location
 
+    @mock.patch("marquez_client.client.MarquezClient._put")
+    def test_create_job_with_run_id(self, mock_put):
+        run_id = str(uuid.uuid4())
+        job_name = "my-job"
+        input_dataset = [
+            {
+                "namespace": "my-namespace",
+                "name": "public.mytable"
+            }
+        ]
+        output_dataset = {
+            "namespace": "my-namespace",
+            "name": "public.mytable"
+        }
+
+        location = "https://github.com/my-jobs/blob/" \
+                   "07f3d2dfc8186cadae9146719e70294a4c7a8ee8"
+
+        context = {
+            "sql": "SELECT * FROM public.mytable;"
+        }
+
+        mock_put.return_value = {
+            "id": {
+                "namespace": "my-namespace",
+                "name": "my-job"
+            },
+            "type": "BATCH",
+            "name": "my-job",
+            "createdAt": "2020-08-12T07:30:55.321059Z",
+            "updatedAt": "2020-08-12T07:30:55.333230Z",
+            "namespace": "my-namespace",
+            "inputs": [
+                {
+                    "namespace": "my-namespace",
+                    "name": "public.mytable"
+                }
+            ],
+            "outputs": [
+                {
+                    "namespace": "my-namespace",
+                    "name": "public.mytable"
+                }
+            ],
+            "location": "https://github.com/my-jobs/blob/"
+                        "07f3d2dfc8186cadae9146719e70294a4c7a8ee8",
+            "context": {
+                "sql": "SELECT * FROM public.mytable;"
+            },
+            "description": "My first job.",
+            "latestRun": None
+        }
+
+        response = self.client.create_job(
+            namespace_name=_NAMESPACE,
+            job_name=job_name,
+            job_type=JobType.BATCH,
+            location=location,
+            input_dataset=input_dataset,
+            output_dataset=output_dataset,
+            context=context,
+            run_id=run_id
+        )
+
+        mock_put.assert_called_once()
+        assert mock_put.call_args.kwargs.get("payload").get("runId")\
+               == run_id
+
+        assert str(response['inputs']) is not None
+        assert response['latestRun'] is None
```

Comment:
The `latestRun` wouldn't be `None` is this case. When a run ID is provided, the Marquez API assumes the job has a run in a `RUNNING` state. We'd want to make sure the contract is correct and mock out a latest run object in the response using the run ID passed to Marquez.

The main goal for adding run ID support is to allow the caller to modify a running job while linking the new job version to the existing run. As an example, you can reference the java client test we have [`testCreateJobWithRunId()`](https://github.com/MarquezProject/marquez-java/blob/feature/add-run-id-to-job-meta/src/test/java/marquez/client/MarquezClientTest.java#L560)

Reply:
i agree with the main goal.